mlrun/api/utils/clients/log_collector.py

```diff
@@ -203,12 +204,32 @@ async def has_logs(
         :param raise_on_error: Whether to raise an exception on error
         :return: Whether the log collector service has logs for the given run
         """
+
+        def _retryable_error(error_message: str) -> bool:
+            retryable_error_regex = [
```

Comment:
`retryable_error` seems like it might will have more use in the future, I would suggest defining it as a separate method, maybe even in a log collector helpers.

Reply:
I thought so too initially, but we won't necessarily want to retry on the same errors for all methods.
E.g - don't retry on `readdirent` errors in `start_log`, because it is not expected at all.